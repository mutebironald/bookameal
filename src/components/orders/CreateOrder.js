import React from 'react';
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import "../.././App.css"

import  addOrderRequest  from '../../actions/addOrderRequest';

import Notifications, {notify} from 'react-notify-toast';
import toastr from 'toastr';

export class CreateOrder extends React.Component {
    constructor(props){
    super(props);
    this.state = {
        menu_id : this.props.id,

        }
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    }

    onChange(event) {
        this.setState({ [event.target.name]: event.target.value});
    }

    onSubmit(event){
        event.preventDefault();
        
            console.log(this.state);
            let  data = {
                'menu_id': this.state.menu_id
            }
            this.props.addOrderRequest((data))
            .then(response => {
                if(response.addOrder){
                    toastr.success(response.message)
                }
                else{
                    toastr.error(response.message)
                }
            })

        this.setState({
            menu_id: ''
        })
    }
    render(){
        return (
            <form onSubmit={this.onSubmit}>
            <Notifications/>

                <div className="form-group">
                    <label className="control-label">Menu Id</label>
                    <input
                        value={this.state.menu_id}
                        onChange={this.onChange}
                        type="number"
                        name="cow"
                        className="form-control"
                    />
                </div>

                <div className="form-group">
                    <button className="btn btn-primary btn-success btn-sm">
                        Create Order
                    </button>
                </div>

            </form>
        );
    }
}
CreateOrder.propTypes = {
    addOrderRequest: PropTypes.func.isRequired
}


export default connect (null, {addOrderRequest})(CreateOrder);
