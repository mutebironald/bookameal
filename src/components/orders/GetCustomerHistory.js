import React from 'react';
import '../.././index.css';
import { Card, CardTitle, CardText } from 'reactstrap';

class GetOrderHistory extends React.Component{
    constructor(props){
        super(props);
        this.state ={
            order: []
        }
    }
    returnOrder = () => {
        const order = this.state.order;
        if(order.data === undefined){
            return (
                <div>
                    <h2 class="text-white">No orders yet</h2>
                </div>
            );
        }
        else {
            return (
                
                order.data.Orders.map((order) => (
                    <div>
                        <Card body outline color="sucess" style={{ width: "28rem", height:"9rem", margin: "10px"}} >
                            <CardTitle className="text-center">Order</CardTitle>
                            <CardText className="text-center"><strong>Meal:</strong>{order.meal}</CardText>
                            <CardText className="text-center"><strong>Time:</strong>{order.order_time}</CardText>
                        </Card>

                    </div>
                    
                ))
            )
        }
    }

    render(){
        return(
            <div>
                {this.returnOrder()}
            </div>
        )
    }



}

export default GetOrderHistory;
