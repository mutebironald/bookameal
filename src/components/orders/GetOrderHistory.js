import React from 'react';
import { Card, CardText } from 'reactstrap';
import '../.././index.css';
import instance from '../../actions/instance';

export class GetOrderHistory extends React.Component{
    constructor(props){
        super(props);
        this.state ={
            order: []
        }
    }
    
    componentDidMount(){
        instance.get(`api/v1/orders/users`, {headers:{"Authorization": localStorage.getItem('Authorization')}})
        .then(res => res)
        .then(resData => {
            let meals = resData
            this.setState({order: meals });
        })
    };


    returnOrder = () => {
        const order = this.state.order;
        if(order.data === undefined){
            return (
                <div>
                    <h2 class="text-white">No orders yet</h2>
                </div>
            );
        }
        else {
            return (
                
                order.data.Orders.map((order) => (
                    <div>
                        <Card body outline color="sucess" style={{ width: "28rem", height:"10rem", margin: "10px"}} >
                            <CardText className="text-center"><strong>Meal:</strong>{order.meal}</CardText>
                            <CardText className="text-center"><strong>Price:</strong>{order.price}</CardText>
                            <CardText className="text-center"><strong>Time:</strong>{order.order_time}</CardText>
                        </Card>

                    </div>
                    
                ))
            )
        }
    }

    render(){
        return(
            <div>
                {this.returnOrder()}
            </div>
        )
    }



}

export default GetOrderHistory;
