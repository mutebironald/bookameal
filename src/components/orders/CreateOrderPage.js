import React, { Component  } from "react";
import CreateOrder from "./CreateOrder";

export class CreateOrderPage extends Component{
    render(){
        return(
            <div className="row">
                <div className="col-md-4 col-md-offset-4">
                

                <CreateOrder />

                </div>
            </div>
        );
    }
}

export default CreateOrderPage;
