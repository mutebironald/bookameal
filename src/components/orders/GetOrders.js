import React from 'react';
import '../.././index.css';
import { Card, Button, CardTitle, CardText } from 'reactstrap';

import CreateOrder from './CreateOrder';

import { Modal, ModalHeader, ModalBody} from 'reactstrap'

import {
        Collapse,
        Navbar,
        NavbarToggler,
        NavbarBrand,
        Nav,
        NavLink,
        } from 'reactstrap';


const selectedStyle = {
        backgroundColor: "white",
        color: "slategasy"
      }

class GetOrders extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            order: [],
            modal: false
            }
            this.toggle = this.toggle.bind(this)
        };

    componentDidMount(){
        fetch('/api/v1/orders')
        .then(response => response.json())
        .then(resData => {

            let meals = resData
           this.setState({ orderMeals: meals });
           console.log("hammingly wathonic",this.state)
        })
    };

    componentWillMount(){
        // console.log("gvduidwfghjf")
        console.log(this.state)
    }

    toggle(){
        this.setState({
            modal: !this.state
        })
    }

    returnOrder = () =>{
        const  order = this.state;

        if(order.orderMeals === undefined){
            console.log(order)
            return (
                <div>
               <h2 class="text-white">No orders yet</h2>
                </div>
            );
        }else {
           return (
           
           order.orderMeals.orders.map((order) =>(
                
                <div>

                <div>
                </div>

        <div>
        <Card body outline color="sucess" style={{ width: "32rem", height:"12rem", margin: "10px"}} >
            <CardTitle className="text-center">Order</CardTitle>
            <CardText className="text-center"><strong>Id:</strong>{order.id}</CardText>
            <CardText className="text-center"><strong>Time:</strong>{order.order_time}</CardText>
        </Card>
        
        </div>

            <Button className="btn btn-success btn-sm" onClick={this.toggle} style={{ width: "4rem", height:"2rem"}}>Add</Button>
            <Modal isOpen={this.state.modal} backdropTransition={{ timeout: 1300 }}
            toggle={this.toggle} className={this.props.className}>
                <ModalHeader toggle={this.toggle}>Menu</ModalHeader>
                <ModalBody>
                    <CreateOrder/>      
                </ModalBody>
            </Modal>



        </div>
            )));
            
        }
    }
    render() {
        return (
            <div>
                <Navbar color="success" light expand="md">
                    <NavbarBrand href="/">Book-A-Meal | Home </NavbarBrand>
                    <NavbarToggler onClick={this.toggle}/>
                    <Collapse isOpen={this.state.isOpen} navbar>
                        <Nav className="ml-auto" navbar>
                        <NavLink href='/getMeals' activeStyle={selectedStyle}>Meals</NavLink>
                        <NavLink href='/getMenu' activeStyle={selectedStyle}>Menu</NavLink>           
                        <NavLink href='/getOrder' activeStyle={selectedStyle}>Orders</NavLink>
                        </Nav>      
                    </Collapse>
                </Navbar>
                {this.returnOrder()}

                

                    {/* <Button className="btn btn-success btn-sm" onClick={this.toggle} style={{ width: "4rem", height:"2rem"}}>Add</Button>
                    <Modal isOpen={this.state.modal} backdropTransition={{ timeout: 1300 }}
                    toggle={this.toggle} className={this.props.className}>
                        <ModalHeader toggle={this.toggle}>Menu</ModalHeader>
                        <ModalBody>
                            <CreateOrder/>      
                        </ModalBody>
                    </Modal> */}



                

            </div>
        );
    }
}
export default GetOrders;
