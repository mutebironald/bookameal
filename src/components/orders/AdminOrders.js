import React, { Component } from 'react';
import '../.././index.css';
import { Card, Button, CardTitle, CardText } from 'reactstrap';

import CreateOrder from './CreateOrder';
import axios from 'axios'

import GetCustomerHistory from './GetCustomerHistory';

import {
        Modal, ModalHeader,
        ModalBody,
        ModalFooter,
        Collapse,
        Navbar,
        NavbarToggler,
        NavbarBrand,
        Nav,
        NavLink
        } from 'reactstrap';


const selectedStyle = {
        backgroundColor: "white",
        color: "slategasy"
      }

class AdminOrders extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            order: [],
            modal: false,
            modal1: false,
            }
            this.toggle = this.toggle.bind(this)
            this.toggle1 = this.toggle1.bind(this)
        };

    componentDidMount(){
        fetch('/api/v1/orders')
        .then(response => response.json())
        .then(resData => {

            let meals = resData
           this.setState({ orderMeals: meals });
           console.log(this.state)
        })
    };

    componentWillMount(){
        console.log(this.state)
    }

    toggle(){
        this.setState({
            modal: !this.state
        })
    }

    toggle1(){
        this.setState({
            modal1: !this.state.modal1
        })
        console.log(this.state)
    }

    returnOrder = () =>{
        const  order = this.state;

        if(order.orderMeals === undefined){
            console.log(order)
            return (
                <div>
               <h2 class="text-white">No orders yet</h2>
                </div>
            );
        }else {
           return (
           
           order.orderMeals.orders.map((order) =>(
                
        <div>


        <div style={{ width: "32rem", height:"12rem", margin: "10px"}} className="card mt-2 border-success text-center center">
            <div className="row text-bold " >
                
            <div className="col-md-9 meal-data text-center">
                <strong>Id:</strong>{order.id}
            </div>

            <div className="col-md-9 meal-data text-center">
                <strong>Price:</strong>{order.price}
            </div>

            <div className="col-md-9 meal-data text-center">
                <strong>Time:</strong>{order.order_time}
            </div>

            <div className="col-md-9 meal-data text-center">
                <strong>Ordered By:</strong>{order.user}
            </div>

            <div className="col-md-3 meal-data text-center">
                <span className="btn btn-success btn-sm" onClick={() => this.toggle1()}>Order History</span>
            </div>
        </div>
        <Modal isOpen={this.state.modal1} backdropTransition={{ timeout: 1300}}
        toggle={this.toggle1} className={this.props.className}>
        <ModalHeader toggle={this.toggle1} class="modal">History</ModalHeader>
        <ModalBody class="modal-main">
            {/* <CreateOrder id={this.state.id} toggle={this.toggle1.bind(this)}/> */}
            <GetCustomerHistory />
        </ModalBody>
        <ModalFooter>
            {/* <button onClick={() => this.resetOrder()} className="btn btn-primary">Close</button> */}
        </ModalFooter>
        </Modal>





        </div>
        </div>
            )));
            
        }
    }
    render() {
        return (
            <div>
                <Navbar color="success" light expand="md">
                    <NavbarBrand href="/">Book-A-Meal | Home </NavbarBrand>
                    <NavbarToggler onClick={this.toggle}/>
                    <Collapse isOpen={this.state.isOpen} navbar>
                        <Nav className="ml-auto" navbar>
                        <NavLink href='/getMeals' activeStyle={selectedStyle}>Meals</NavLink>
                        <NavLink href='/adminMenu' activeStyle={selectedStyle}>Menu</NavLink>           
                        <NavLink href='/adminOrders' activeStyle={selectedStyle}>Orders</NavLink>
                        <NavLink href='/adminlogin' activeStyle={selectedStyle}>Logout</NavLink>
                        </Nav>      
                    </Collapse>
                </Navbar>
                {this.returnOrder()} 

            </div>
        );
    }
}
export default AdminOrders;
