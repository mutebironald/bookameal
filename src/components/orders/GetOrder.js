import React, {Component} from 'react';
import '../.././index.css';
import { Card, Button, CardTitle, CardText } from 'reactstrap';
import GetOrders from './GetOrders';

class GetOrders extends React.Component{
    
}
