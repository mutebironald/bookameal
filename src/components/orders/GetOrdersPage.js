import React, { Component  } from "react";
import GetOrders from "./GetOrders";
class GetOrdersPage extends Component{
    render(){
        return(
            <div className="card mt-1">
                <div className="row text-muted">
                    <div className="col-md-3 meal-data">
                    <strong>Order:</strong>{this.props.item.id}
                    </div>
                    <div className="col-md-5 meal-data">
                    <strong>Name:</strong>{this.props.item.name}
                    </div>
                    <div className="col-md-3 p-2">
                    <a href="/createOrder" id = {this.props.item.id} className="btn btn-info btn-sm" role="button" aria-pressed="true">Edit Order</a>
                    <br/><br/>
                    <button
                        className="btn btn-danger">
                        delete
                    </button>
                    </div>
                </div>
            </div>

        )
    }
}

export default GetOrdersPage;
