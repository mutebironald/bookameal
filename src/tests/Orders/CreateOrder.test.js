import React from 'react';
import { CreateOrder } from '../../components/orders/CreateOrder';




describe("Create order component", () => {
    let wrapper;
    let mockFn;
    beforeEach(() => {
        mockFn = jest.fn();
        wrapper = shallow(<CreateOrder addOrderRequest = {() =>Promise.resolve({addOrder:true})}/>)
    });


    it("should render correctly", ()=> {
        expect(wrapper).toMatchSnapshot();
    });

    it("should submit", () => {
        const evt = {
            preventDefault: mockFn
        }
        wrapper.instance().onSubmit(evt);
        expect(mockFn.mock.calls.length).toBe(1);
    });

})
