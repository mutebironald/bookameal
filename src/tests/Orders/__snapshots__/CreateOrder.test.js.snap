// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create order component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateOrder
    addOrderRequest={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <_class
          options={Object {}}
/>,
        <div
          className="form-group"
>
          <label
                    className="control-label"
          >
                    Menu Id
          </label>
          <input
                    className="form-control"
                    name="cow"
                    onChange={[Function]}
                    type="number"
                    value={undefined}
          />
</div>,
        <div
          className="form-group"
>
          <button
                    className="btn btn-primary btn-success btn-sm"
          >
                    Create Order
          </button>
</div>,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "options": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="control-label"
>
              Menu Id
</label>,
            <input
              className="form-control"
              name="cow"
              onChange={[Function]}
              type="number"
              value={undefined}
/>,
          ],
          "className": "form-group",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Menu Id",
              "className": "control-label",
            },
            "ref": null,
            "rendered": "Menu Id",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "name": "cow",
              "onChange": [Function],
              "type": "number",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn btn-primary btn-success btn-sm"
>
            Create Order
</button>,
          "className": "form-group",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create Order",
            "className": "btn btn-primary btn-success btn-sm",
          },
          "ref": null,
          "rendered": "Create Order",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <_class
            options={Object {}}
/>,
          <div
            className="form-group"
>
            <label
                        className="control-label"
            >
                        Menu Id
            </label>
            <input
                        className="form-control"
                        name="cow"
                        onChange={[Function]}
                        type="number"
                        value={undefined}
            />
</div>,
          <div
            className="form-group"
>
            <button
                        className="btn btn-primary btn-success btn-sm"
            >
                        Create Order
            </button>
</div>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "options": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="control-label"
>
                Menu Id
</label>,
              <input
                className="form-control"
                name="cow"
                onChange={[Function]}
                type="number"
                value={undefined}
/>,
            ],
            "className": "form-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Menu Id",
                "className": "control-label",
              },
              "ref": null,
              "rendered": "Menu Id",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "name": "cow",
                "onChange": [Function],
                "type": "number",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-primary btn-success btn-sm"
>
              Create Order
</button>,
            "className": "form-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create Order",
              "className": "btn btn-primary btn-success btn-sm",
            },
            "ref": null,
            "rendered": "Create Order",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
